woocommerce/woocommerce
# Fix editing Shipping methods auto-saving

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR fixes the issue with Shipping Methods auto-saving the entire form. Now, when a shipping method is added and there is another change in the form, the `Save changes` button will remain enabled, and no other change will be saved except for the added method.

![Screen Capture on 2023-05-26 at 12-27-00](https://github.com/woocommerce/woocommerce/assets/1314156/0a371ffe-9037-4291-bde0-dd174fa8b5cf)

Closes #38114.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go `WooCommerce` > `Settings` > `Shipping` (URL: `wp-admin/admin.php?page=wc-settings&tab=shipping`).
2. Create a shipping zone and click edit.
3. Update the zone name and add a new shipping method
4. Notice how the loading icon shows up on the shipping method table, and the form gets saved without clicking `Save changes`.
5. Add another change to the form (like changing `Zone name`).
6. Add another shipping method.
7. Verify that the `Save changes` button is still enabled and only the added shipping method is saved.

_We currently [don't have a method to delete shipping zones](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-ajax.php#L128-L180) individually, which is why I opted for this workaround.  The form will be saved automatically if we add a new shipping method. However, if another change is made, then pressing `Save changes` will be necessary to save it._

_When editing a specific shipping method, all the changes in the form will be saved after pressing the `Save changes` button inside the modal._
